controllers/tc000121_destroy_on_delete_test.go

```diff
@@ -105,15 +105,15 @@ func Test_000121_destroy_on_delete_test(t *testing.T) {
 			Vars: []infrav1.Variable{
 				{
 					Name:  "kubeconfig",
-					Value: utils.JsonEncodeBytes([]byte(testEnvKubeConfigPath)),
+					Value: utils.MustJSONEncodeBytes([]byte(testEnvKubeConfigPath)),
```

Comment:
I don't think the `Must` idiom is appropriate here?

The string that is being encoded is provided by users, and so, it could be not-encodable as valid JSON, `panic()` can cause the process to crash, which doesn't sound like a desirable outcome?